src/data-types/wrappers.h

```diff
@@ -0,0 +1,326 @@
+#pragma once
```

Comment:
On Thu, Mar 05, 2015 at 01:37:54PM -0800, Hoà V. DINH wrote:
> > @@ -0,0 +1,326 @@
> > +#pragma once
> - Could you use #ifndef / #define / #endif instead?

Too many old compilers? ;-) Yes, of course there may be include guards
instead of the pragma once.

> - Rename the file to syscall_wrappers.h

Why not.

> - Could you add a global flag that would change the behavior of
> Fopen() to just call fopen() (and not do the loop) and same for other
> syscalls?

This would be useless. First: the wrappers are written in the way that
if an operating system does not deliver EINTR (like Windoze), they
behave exactly like that already. If at a certain place you don't want
to have the wrapper in your code, you could just use the original
function (which is the reason why wrappers aren't called by using a
#define to modify the original names).

Yours,
VB.
-- 
Volker Birk
Oberer Graben 4, 8400 Winterthur, Schweiz
mailto:vb@dingens.org  http://fdik.org